src/Internal/Test/ContractTest.purs

```diff
@@ -60,6 +74,30 @@ newtype ContractTestPlan = ContractTestPlan
     -> r
   )
 
+instance Semigroup ContractTestPlan where
+  append
+    (ContractTestPlan runContractTestPlan)
+    (ContractTestPlan runContractTestPlan') =
+    do
+      runContractTestPlan \distr tests -> do
+        runContractTestPlan'
+          \distr' tests' -> ContractTestPlan \h -> h (distr /\ distr') do
+            mapTest (_ <<< fst) tests
+            mapTest (_ <<< snd) tests'
```

Comment:
I don't think this is a valid semigroup: `/\` is not associative. From the point of view of the user, sure. But not *actually*. We are using it only in one place, so please make it a function.